junit-runtime/src/main/scala/org/scalajs/junit/JUnitExecuteTest.scala

```diff
@@ -0,0 +1,227 @@
+package org.scalajs.junit
+
+import com.novocode.junit.Ansi._
+import com.novocode.junit.{OutputCapture, RichLogger}
+import org.junit._
+
+import scala.util.{Failure, Success, Try}
+
+final class JUnitExecuteTest(fullyQualifiedName: String,
+    runner: JUnitBaseRunner, classMetadata: JUnitTestMetadata,
+    richLogger: RichLogger) {
+
+  def executeTests(): Unit = {
+    val packageName = fullyQualifiedName.split('.').init.mkString(".")
+    val className = fullyQualifiedName.split('.').last
+
+    val jUnitMetadata = classMetadata.scalajs$junit$metadata
+
+    Try {
+      for (method <- jUnitMetadata.beforeClassMethod)
+        classMetadata.scalajs$junit$invoke(method.id)
+    } match {
+      case Success(_) =>
+        val outputCapture =
+          if (runner.runSettings.quiet) OutputCapture.start()
+          else null
+
+        try {
+          for (method <- jUnitMetadata.testMethods) {
+            method.getIgnoreAnnotation match {
+              case Some(ign) =>
+                ignoreTest(packageName, className, method.name)
+
+              case None =>
+                executeTestMethod(classMetadata, packageName, className, method)
+            }
+          }
+        } finally {
+          if (outputCapture != null)
+            outputCapture.stop()
+        }
+
+        for (method <- jUnitMetadata.afterClassMethod)
+          classMetadata.scalajs$junit$invoke(method.id)
+
+      case Failure(ex: AssumptionViolatedException[_]) =>
+        logFormattedInfo(packageName, className, null, "ignored")
+        runner.taskSkipped()
+
+      case Failure(ex: internal.AssumptionViolatedException) =>
+        logFormattedInfo(packageName, className, null, "ignored")
+        runner.taskSkipped()
+
+      case Failure(ex) => throw ex
+    }
+  }
+
+  private[this] def executeTestMethod(classMetadata: JUnitTestMetadata,
+      packageName: String, className: String, method: JUnitMethodMetadata) = {
+    val jUnitMetadata = classMetadata.scalajs$junit$metadata
+    val testClassInstance = classMetadata.scalajs$junit$newInstance
+
+    val t0 = System.currentTimeMillis
```

Comment:
Consider `System.nanoTime` instead. `currentTimeMillis` is not guaranteed to be monotonic, and is therefore not suited for measuring elapsed time.